docs/How_config_files_are_merged.md

```diff
@@ -0,0 +1,576 @@
+```
+
+## Best Practices
+
+If you choose to use only global configuration files or project-level configuration files it will be easier to remember and find the set of profile properties that will be used by CLI commands. Combining both a global and project configuration works in a consistent fashion, but you must remember that your current directory dictates when one or both configurations apply, and you must remember the different rules that are used when merging between global and project configurations.
+
+### **Advantages of only using a global configuration**
+
+You might choose to use a global configuration file if you typically use the same set of hosts and services for all of your mainframe activities. You will only maintain one Zowe CLI configuration that will be used for CLI commands regardless of the directory in which you are located when you run a zowe command. You can edit your global configuration file in a text editor to make quick temporary changes. For example, you might briefly change the port number of a service to try out a test instance of a given service. Similarly, you could quickly change the default profile for a given profile type with a quick change in your text editor. By doing that, you could perform the same operation against different mainframe hosts with the exact same zowe command, rather than having to specify a different profile name option on each command.
+
+The config file's ability to understand comments (// or /* */) can make such quick edits even easier. You can comment-out several alternatives and then just uncomment the one you want to use, rather than having to remember the correct value and retyping that alternative.
+
+Since a global configuration exists under your HOME directory, it belongs to you personally. Thus, you may not have a great need for a user configuration file in this scenario - all of the values in a global configuration already belong to you. If you want to periodically share your configuration with colleagues, you could choose to place your standard configuration in a normal team configuration file (zowe.config.json) and place your specialty alternatives in your user configuration (zowe.config.user.json). If you receive an ad-hoc request from a colleague for your configuration, you could just send your zowe.config.json file to that person. Thus you could avoid scrubbing your configuration file before sending it to someone else.
+
+### **Advantages of only using a project configuration**
```

Comment:
Another potential advantage is that you can easily switch between v1 and v2 profiles depending on which directory you run `zowe` commands in. I'll defer to you on whether this is worth mentioning 🙂

Reply:
It is both an advantage and a potential point of confusion, so it is worth mentioning.